File: lib/genetic_algorithm/mutation_functions/RegularMutationFunction.py
import random
from lib.genetic_algorithm.mutation_functions.MutationFunction import MutationFunction


class RegularMutationFunction(MutationFunction):
    """
    Mutation Function class that mutates an individual given a mutation probability
    """

    def __init__(self, mutation_prob):
        """
        :param mutation_prob: <float> the probability of any indivual to mutate
        """
        self.mutation_prob = mutation_prob

    def mutate(self, individual):
        """
        mutates an individual, given the probability
        :param individual: <InputIndividual> the individual to perform mutation on
        """
        if random.random() < self.mutation_prob:
            individual.mutate()

File: love_ga_wrapper/love_ga_wrapper/lib/genetic_algorithms/mutation_functions/RegularMutationFunction.lua
local class = require("love_ga_wrapper.lib.classes.class")
local extend = require("love_ga_wrapper.lib.classes.extend")
local MutationFunction =  require("love_ga_wrapper.lib.genetic_algorithms.mutation_functions.MutationFunction")
--------------------------------------------------------------------------------------------------------

-- class: RegularMutationFunction
-- param: mutation_prob:int -> the probability of any indivual to mutate
-- Mutation Function class that mutates an individual given a mutation probability
local RegularMutationFunction = extend(MutationFunction, function(self, mutation_prob)
    self.mutation_prob = mutation_prob
end)

-- mutate: Individual -> None
-- mutates an individual with the functions given probability
function RegularMutationFunction.mutate(self, individual)
    if math.random() < self.mutation_prob then
        individual:mutate()
    end
end

return RegularMutationFunction
